File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/ReviewVerificationsActivity.java
package com.workforcex.employer.ui.employer;

import android.os.Bundle;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import com.workforcex.employer.api.RetrofitClient;
import com.workforcex.employer.api.Verification;
import com.workforcex.employer.databinding.ActivityReviewVerificationsBinding;
import com.workforcex.employer.utils.TokenManager;
import java.util.List;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ReviewVerificationsActivity extends AppCompatActivity {

    private ActivityReviewVerificationsBinding binding;
    private TokenManager tokenManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityReviewVerificationsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        tokenManager = new TokenManager(this);

        setTitle("Review Verifications");
        binding.rvVerifications.setLayoutManager(new LinearLayoutManager(this));
        loadPendingVerifications();
    }

    private void loadPendingVerifications() {
        binding.progressBar.setVisibility(View.VISIBLE);
        RetrofitClient.get().getPendingVerifications(tokenManager.getBearerToken())
                .enqueue(new Callback<List<Verification>>() {
                    @Override
                    public void onResponse(Call<List<Verification>> call, Response<List<Verification>> response) {
                        binding.progressBar.setVisibility(View.GONE);
                        if (response.isSuccessful() && response.body() != null) {
                            binding.rvVerifications.setAdapter(new VerificationAdapter(response.body(), ReviewVerificationsActivity.this::updateVerificationStatus));
                        } else {
                            Toast.makeText(ReviewVerificationsActivity.this, "Failed to load verifications", Toast.LENGTH_SHORT).show();
                        }
                    }

                    @Override
                    public void onFailure(Call<List<Verification>> call, Throwable t) {
                        binding.progressBar.setVisibility(View.GONE);
                        Toast.makeText(ReviewVerificationsActivity.this, "Network error", Toast.LENGTH_SHORT).show();
                    }
                });
    }

    private void updateVerificationStatus(String verificationId, String status, String comments) {
        RetrofitClient.get().updateEmployerVerificationStatus(tokenManager.getBearerToken(), verificationId, status, comments)
                .enqueue(new Callback<Void>() {
                    @Override
                    public void onResponse(Call<Void> call, Response<Void> response) {
                        if (response.isSuccessful()) {
                            Toast.makeText(ReviewVerificationsActivity.this, "Verification status updated", Toast.LENGTH_SHORT).show();
                            loadPendingVerifications(); // Refresh the list
                        } else {
                            Toast.makeText(ReviewVerificationsActivity.this, "Failed to update status", Toast.LENGTH_SHORT).show();
                        }
                    }

                    @Override
                    public void onFailure(Call<Void> call, Throwable t) {
                        Toast.makeText(ReviewVerificationsActivity.this, "Network error", Toast.LENGTH_SHORT).show();
                    }
                });
    }
}

File: android/employer/app/src/main/java/com/workforcex/employer/ui/employer/ReviewVerificationsActivity.kt
package com.workforcex.employer.ui.employer

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.workforcex.employer.databinding.ActivityReviewVerificationsBinding
import com.workforcex.employer.utils.TokenManager
import com.workforcex.shared_employer.RetrofitClient
import com.workforcex.shared_employer.models.Verification
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class ReviewVerificationsActivity : AppCompatActivity() {

    private lateinit var binding: ActivityReviewVerificationsBinding
    private lateinit var tokenManager: TokenManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityReviewVerificationsBinding.inflate(layoutInflater)
        setContentView(binding.root)
        tokenManager = TokenManager(this)

        title = "Review Verifications"
        binding.rvVerifications.layoutManager = LinearLayoutManager(this)
        loadPendingVerifications()
    }

    private fun loadPendingVerifications() {
        binding.progressBar.visibility = View.VISIBLE
        RetrofitClient.get().getPendingVerifications(tokenManager.getBearerToken())
            .enqueue(object : Callback<List<Verification>> {
                override fun onResponse(
                    call: Call<List<Verification>>,
                    response: Response<List<Verification>>
                ) {
                    binding.progressBar.visibility = View.GONE
                    if (response.isSuccessful && response.body() != null) {
                        binding.rvVerifications.adapter =
                            VerificationAdapter(response.body()!!)
                    } else {
                        Toast.makeText(this@ReviewVerificationsActivity, "Failed to load verifications", Toast.LENGTH_SHORT).show()
                    }
                }

                override fun onFailure(call: Call<List<Verification>>, t: Throwable) {
                    binding.progressBar.visibility = View.GONE
                    Toast.makeText(this@ReviewVerificationsActivity, "Network error", Toast.LENGTH_SHORT).show()
                }
            })
    }
}
